src/compas_rhino/geometry/brep/brep.py

```diff
@@ -0,0 +1,248 @@
+from compas.geometry import Frame
+from compas.geometry import Brep
+from compas.geometry import BrepInvalidError
+from compas.geometry import BrepTrimmingError
+
+from compas_rhino.conversions import box_to_rhino
+from compas_rhino.conversions import point_to_rhino
+from compas_rhino.conversions import xform_to_rhino
+from compas_rhino.conversions import frame_to_rhino
+
+import Rhino
+
+from .face import RhinoBrepFace
+from .edge import RhinoBrepEdge
+from .vertex import RhinoBrepVertex
+from .loop import RhinoBrepLoop
+
+
+TOLERANCE = 1e-6
+
+
+class RhinoBrep(Brep):
+    """
+    Rhino Brep backend class.
+    Wraps around and allows serialization and de-serialization of a `Rhino.Geometry.Brep`
+    """
+
+    # this makes de-serialization backend-agnostic.
+    # The deserializing type is determined by plugin availability when de-serializing
+    # regardless of the context available when serializing.
+    __class__ = Brep
+
+    def __new__(cls, *args, **kwargs):
+        # This breaks the endless recursion when calling `compas.geometry.Brep()` and allows
+        # having Brep here as the parent class. Otherwise RhinoBrep() calls Brep.__new__()
+        # which calls RhinoBrep() and so on...
+        return object.__new__(cls, *args, **kwargs)
+
+    def __init__(self, brep=None):
+        super(RhinoBrep, self).__init__()
+        self._brep = brep or Rhino.Geometry.Brep()
+
+    # ==============================================================================
+    # Constructors
+    # ==============================================================================
+
+    @classmethod
+    def from_brep(cls, rhino_brep):
+        brep = cls(rhino_brep)
+        return brep
+
+    @classmethod
+    def from_box(cls, box):
+        rhino_box = box_to_rhino(box)
+        return cls.from_brep(rhino_box.ToBrep())
+
+    # ==============================================================================
+    # Data
+    # ==============================================================================
+
+    @property
+    def data(self):
+        faces = []
+        for face in self.faces:
+            faces.append(face.data)
+        return {"faces": faces}
+
+    @data.setter
+    def data(self, data):
+        """
+        Parameters
+        ----------
+        data
+
+        Returns
+        -------
+
+        """
+        faces = []
+        for facedata in data["faces"]:
+            face = RhinoBrepFace.from_data(facedata)
+            faces.append(face)
+        self._create_native_brep(faces)
+
+    # ==============================================================================
+    # Properties
+    # ==============================================================================
+
+    @property
+    def native_brep(self):
+        """
+        Returns the native representation of this Brep.
+
+        Returns
+        -------
+        :class: `Rhino.Geometry.Brep`
+        """
```

Comment:
any reason some of the properties and others not? perhaps leave this one out as well?

Reply:
You mean why there's public access to the native type in Brep but not in the others? 

I found this is useful at the moment due to the lack of a `BrepArtist` for example. That way you can plug-in `Brep.native_brep` to a GH geometry component to visualize.

Maybe once there's more extensive support for Brep in COMPAS it would make sense to remove this one as well.

Reply:
No particular reason ;)
Removed all docstrings from properties and added to Attributes instead.